doc/docusaurus/docs/using-plinth/evaluating-plinth.md

```diff
@@ -0,0 +1,201 @@
+---
+sidebar_position: 12
+---
+
+# Evaluating CompiledCode for testing
+
+The `CompiledCode` is intended to be evaluated by Cardano nodes when 
+transaction validation occurs. For this purpose, it is serialized and included in a transaction.
+
+However, it is also possible to evaluate `CompiledCode` in a test environment. 
+This is useful for testing and troubleshooting. By doing so, Plinth developers can
+not only get the immediate result of the code but also obtain the traces emitted 
+during the evaluation, as well as the consumed execution budget.
+
+Let's consider the following example Plinth program:
+<LiteralInclude 
+  file="Example/Evaluation/Main.hs" 
+  language="haskell" 
+  title="Example Plinth code" 
+  start="-- BEGIN Plinth" 
+  end="-- END Plinth" />
+
+In order to evaluate this code, we need to add the `plutus-tx:plutus-tx-testlib` 
+dependency to our cabal file:
+```cabal
+  build-depends:
+    , plutus-tx:plutus-tx-testlib
+```
+
+So that we can import the necessary functionality:
+
+```haskell
+import PlutusTx.Test (
+  CodeResult,
+  applyLifted,
+  evaluateCompiledCode,
+  prettyCodeResult
+ )
+```
+
+Our example code represents a function that expects an `Integer` argument
+and returns the `Integer` value. 
+
+It is possible to evaluate this compiled code without applying it to any arguments, 
+and evaluation will succeed, returning the value of type `Integer -> Integer`:
+
+```haskell
+result :: CodeResult
+result = evaluateCompiledCode compiledCode
```

Comment:
The cost model and parameters depend on `MajorProtocolVersion` and ledger language version (V1, V2, V3), so I'd expect `evaluateCompiledCode`  to take them as parameters. It could also take arguments to `compiledCode` as a parameter.

Reply:
I see a trade-off here: on one hand I want to expose the most simple API to the users, and the less decisions they have to make - the better it is. On the other - we want it to be flexible: e.g. users can decide which CEK Parameters to use, but it means that CEK Parameters parameter needs to be exposed. 
I decided to hardcode the `defaultCekParametersForTesting` thus limiting the function applicability to testing. 
From your comment I read that you'd go different route: ask users to chose the CEK parameters, directly or indirectly. I am thinking maybe we can have two functions for both sides of the trade-off...